modules/core/shared/src/main/scala/gem/math/Angle.scala

```diff
@@ -227,17 +239,91 @@ object Angle extends AngleOptics {
 
 }
 
-trait AngleOptics { this: Angle.type =>
+trait AngleOptics extends OpticsHelpers { this: Angle.type =>
+
+  /**
+   * Microarcseconds, in [0, 360°).
+   * @group Optics
+   */
+  lazy val microarcseconds: SplitMono[Angle, Long] =
+    SplitMono(_.toMicroarcseconds, Angle.fromMicroarcseconds)
+
+  /**
+   * Signed microarcseconds, in [-180°, 180°).
+   * @group Optics
+   */
+  lazy val signedMicroarcseconds: SplitMono[Angle, Long] = {
+    lazy val µas360 = Angle.Angle180.toMicroarcseconds * 2L
+    microarcseconds.imapB[Long](
+      identity,
+      µas => if (µas >= Angle.Angle180.toMicroarcseconds) µas - µas360  else µas
+    )
+  }
+
+  /**
+   * Milliarcseconds, in [0, 360°).
+   * @group Optics
+   */
+  lazy val milliarcseconds: Wedge[Angle, Int] =
+    microarcseconds.scaled(1000L)
+
+  /**
+   * Arcseconds, in [0, 360°).
+   * @group Optics
+   */
+  lazy val arcseconds: Wedge[Angle, Int] =
+    microarcseconds.scaled(1000L * 1000L)
+
+  /**
+   * Arcminutes, in [0, 360°).
+   * @group Optics
+   */
+  lazy val arcminutes: Wedge[Angle, Int] =
+      microarcseconds.scaled(1000L * 1000L * 60L)
 
-  val hourAngle: Prism[Angle, HourAngle] =
-    Prism((_: Angle).toHourAngleExact)(_.toAngle)
+  /**
+   * Degrees, in [0, 360).
+   * @group Optics
+   */
+  lazy val degrees: Wedge[Angle, Int] = microarcseconds.scaled(1000L * 1000L * 60L * 60L)
 
-  val fromStringDMS: Format[String, Angle] =
-    Format(AngleParsers.dms.parseExact, _.toDMS.format)
+  /**
+   * This angle, rounded down to the nearest HourAngle.
+   * @group Optics
+   */
+  lazy val hourAngle: SplitEpi[Angle, HourAngle] =
+    SplitEpi(a => HourAngle.microseconds.reverseGet(a.toMicroarcseconds.toLong / 15L), identity)
 
-  val fromStringSignedDMS: Format[String, Angle] =
+  /**
+   * This angle as an HourAngle, where defined.
+   * @group Optics
+   */
+  lazy val hourAngleExact: Prism[Angle, HourAngle] =
```

Comment:
Here too, why is it a `Prism` instead of `SplitEpi`?

Reply:
There are two here. `hourAngle` is a `SplitEpi` that rounds to the nearest 15 µas, so it's losing information: it provides `Angle => HourAngle`.

`hourAngleExact` only succeeds if the angle is divisible by 15 µas, so it's not defined in cases where information would be lost. It provides `Angle => Option[HourAngle]`.